src/panoptes/utils/horizon.py

```diff
@@ -1,18 +1,105 @@
 import numpy as np
+from astropy import units as u
+
+from panoptes.utils.utils import get_quantity_value
+
+# Implicit variable used to indicate no obstruction at a given az
+NO_HORIZON = None
+
+
+class Obstruction(object):
+
+    def __init__(self, points_list):
+        """ An obstruction is defined by a list of alt, az pairs in clockwise ordering.
+        Args:
+            points_list (list): The list of points, e.g. [[alt1, az1], [alt2, az2]].
+        """
+        super().__init__()
+
+        alt_list = []
+        az_list = []
+
+        if len(points_list) < 2:
+            raise ValueError("Need at least two points for obstruction.")
+
+        for p in points_list:
+            if len(p) != 2:
+                raise ValueError("points_list must be provided as alt/az pairs.")
+
+            alt = get_quantity_value(p[0], u.deg)
+            az = get_quantity_value(p[1], u.deg)
+
+            if az < 0:
+                az += 360
+
+            if abs(alt) > 90:
+                raise ValueError("Altitudes must be between ±90 deg.")
```

Comment:
Github tells me this line is not getting covered by tests and it should be pretty easy to add to up the coverage.